lib/syskit/cli/log_runtime_archive_main.rb

```diff
@@ -61,6 +61,42 @@ def archive(root_dir, target_dir)
                 archiver.process_root_folder
             end
 
+            desc "watch_transfer", "watches a dataset root folder \
+                                    and periodically performs transfer"
+            option :period,
+                   type: :numeric, default: 600, desc: "polling period in seconds"
+            option :max_size,
+                   type: :numeric, default: 10_000, desc: "max log size in MB"
+            default_task def watch_transfer(src_dir, tgt_dir, server_params)
+                loop do
+                    begin
+                        transfer(src_dir, tgt_dir, server_params)
+                    rescue Errno::ENOSPC
+                        next
+                    end
+
+                    puts "Transferred pending logs, sleeping #{options[:period]}s"
+                    sleep options[:period]
+                end
+            end
+
+            desc "transfer", "transfers the datasets"
+            option :max_size,
+                   type: :numeric, default: 10_000, desc: "max log size in MB"
+            def transfer(src_dir, tgt_dir, server_params)
```

Comment:
src_dir -> base_log_dir

Reply:
tgt_dir will become the last folder inside the path for each dataset. It will be available inside process_transfer, when you are iterating on the dataset candidates, each candidate path (which is the timestamp in our case) is the tgt dir